spec/controllers/api/braintree_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+describe Api::BraintreeController do
+
+  # endpoint /api/braintree/token
+  #
+  describe "GET token" do
+    before do
+      allow(::Braintree::ClientToken).to receive(:generate){ '1234' }
+
+      get :token
+    end
+
+    it 'fetches token from braintree' do
+      expect(::Braintree::ClientToken).to have_received(:generate)
+    end
+
+    it 'renders json' do
+      expect(response.body).to eq( {token: '1234'}.to_json )
+    end
+  end
+
+  describe "POST transaction" do
+
+    let(:params) do {
+      payment_method_nonce: 'fake-valid-nonce',
+      amount: '100',
+      user: {
+        first_name: 'George',
+        last_name: 'Orwell',
+        email:'big@brother.com',
+        }
+    }
+    end
+
+    context "valid transaction" do
+
+      let(:sale_object){ double(:sale, success?: true, transaction_id: '1234') }
+
+      before do
+        # Sketching out the API, so don't care what PaymentProcessor::Clients::Braintree::Transaction#make_transaction
+        # does at the moment.
+        # Just want to make sure it's being called.
+        #
+        allow(PaymentProcessor::Clients::Braintree::Transaction).to receive(:make_transaction){ sale_object }
```

Comment:
This method stub on `make_transaction` returns `sale_object`, causing this expectation to fail:

`expect(response.body).to eq( { success: true, transaction_id: '1234' }.to_json )`

The controller action is rendering the response from the braintree wrapper:

`render json: braintree.make_transaction(options)`

But, because of the stubbed method above, we're actually rendering the `sale_object`, which is an `rspec` test double...

`render json: sale_object`

Strangely, `{"__expired"=>false, "name"=>"sale"}` is what we get when we call `render` with `json: sale_object`.